### ISSUE TITLE
Canadian National Calendar

### ISSUE THREAD

**Issue by chrissherren** [Liturgical-Calendar/LiturgicalCalendarAPI#266]
Hello from Canada!
Thanks for starting this project!

I'd like to help by providing the Canadian national calendar. I've attached a json in English. I did not include the common or the readings yet but I will update it later to include those properties. I was focusing on getting the feasts to at least show up in the calendar on the correct days first.

[CANADA.json](https://github.com/user-attachments/files/17903841/CANADA.json)

Of course, in Canada we also have a French missal. It has all the same feasts on the same days so it is just a matter of including the translation. I have the French missal at my office so I can do that. Would I use the json file: nations/Americas/fr_CA.json to create this? I assume each object just needs to have a tag matching it to the Canadian English calendar and the name property with the French name for the fest. Is that correct? Or do I need to include the action property in metadata for each object in fr_CA.json ?

For reference, the Canadian Conference of Catholic Bishops have posted our calendar online in both English and French. The Links below will take you directly to the PDFs.

English: (https://www.cccb.ca/wp-content/uploads/2017/09/nationallitcalender2016_01_27.pdf)

French: (https://www.cecc.ca/wp-content/uploads/2017/11/cal_lit_approuve2016_01_27-3.pdf)

**Comment by JohnRDOrazio** [comment]
Thanks a lot for this contribution! I'll try to get it integrated as soon as I can. The format has changed slightly in a newer version of the API that I've been working on, thanks to contributions from some fellow Catholic developers from the Open Source Catholic network, but I can take care of updating the format (basically all of the JSON keys are now lowercase and snake_case, to make it easier for developers).
I haven't yet implemented multiple languages for National Calendars, but I foresaw that this would come up, and I have created an issue for this #150 . So with this contribution, I guess it will be the perfect opportunity to get this worked out in the upcoming release of v4 of the API!

**Comment by JohnRDOrazio** [comment]
The wider region celebration of the Immaculate Conception will be taken care of by the `wider_region` data that is already defined for the Americas, so it can be removed from this JSON file, and any translations taken care of in weblate (I'll have to create a translation component for this)

**Comment by JohnRDOrazio** [comment]
If I understand correctly, comparing to the provided national calendars, Saints Joachim and Anne are celebrated as a Feast only in the dioceses of the Province of Quebec, and not all throughout Canada, so I'll put that aside for now and we can define it in a local diocesan calendar.
```json
        {
            "festivity": {
                "month": 7,
                "day": 26,
                "event_key": "StsJoachimAnne",
                "name": "Saint Anne, Patron of the Province of Quebec and Saint Joachim, Parents of the Blessed Virgin Mary",
                "grade": 4,
                "color": [
                    "white"
                ]
            },
            "metadata": {
                "action": "setProperty",
                "since_year": 1876
            }
        }
```

**Comment by JohnRDOrazio** [comment]
fixed `Saint André Grasset` => `Blessed André Grasset`

**Comment by JohnRDOrazio** [comment]
fixed `StPiusV` from `createNew` to `moveFestivity`, because it already exists in the General Roman Calendar on April 30th. So moving it to May 1st makes sense seeing there is `StMariedelIncarnation` that is created on April 30th.

**Comment by JohnRDOrazio** [comment]
refactored `StsJeanBrebeuf` from `createNew` to `moveFestivity` and then `setProperty`->`grade` to a value of `4`

**Comment by JohnRDOrazio** [comment]
There were a few blockers for fully implementing the national calendar for Canada, which actually brought to life a couple bugs in the API other than become the opportunity to implement some long awaited enhancements:
* issue #150 (fixed in PR #267)
* issue #271 (fixed in PRs #272 and #275)
* issue #273 (fixed in PR #274)

These have now been fixed, so the national calendar for Canada in English should now be working on the development branch.

Only thing missing now is to complete the French translation for Canada.

**Comment by JohnRDOrazio** [comment]
French should now be implemented, thanks to the contributions from Fr. Chris Sherren:
* PR #270 
* PR #276
